### DISCUSSION TITLE
Routing messages from one destination to another

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#5088] [Q&A]
Is it possible, in a javascript writer destination, to send an HL7 message to another destination in the same channel?  I have a message that contains multiple MRNs for the same patient. I need to split these into separate messages and forward them out to another channel.
I can do the splitting\message generation in the javascript writer but would like to just have another destination in the same channel that sends the message out vs having to send it to a 2nd channel that then sends it out.

**Comment by pacmano1** [comment]
Can you provide more details on that use case with some message examples?

**Comment by tonygermano** [comment]
You cannot route a new message directly to a destination. It must process through the channel source first.

**Comment by franklwambo** [comment]
The source connector needs to first receive the message as a file reader, TCP listener, HTTP listener etc.  While in say the first destination, you can retrieve the original message using the function connectorMessage.getRawData()  then do the necessary transformations then finally have the message you wish to send packaged in a channel map then retrieve that channel map variable at the subsequent destination.
